### PR TITLE
Add.`Redux`

### PR DESCRIPTION
**Pull request by epoundor**

### CODE REVIEW

File: apps/frontend/src/reportWebVitals.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ReportHandler } from 'web-vitals';
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
```
Comment (YacineSteeve) [inline apps/frontend/src/reportWebVitals.ts:3]: Are `WebVitals` a mandatory configuration for `redux` ?

File: apps/frontend/src/views/Preview/Preview.tsx [hunk 2]
```diff
@@ -32,6 +34,8 @@ const Preview: React.FunctionComponent = () => {
                     }`}>
                     Drop it here
                 </div>
+
+                {previewTree}
```
Comment (YacineSteeve) [inline apps/frontend/src/views/Preview/Preview.tsx:38]: 😂

File: apps/frontend/src/store/previewTree/previewTreeSlice.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        addComponent: (state) => {
+            return state;
+        },
+        addChildren: (state) => {
+            return;
+        },
+        arrange: (state) => {
+            return;
+        },
+        deleteComponent: (state) => {
```
Comment (YacineSteeve) [inline apps/frontend/src/store/previewTree/previewTreeSlice.ts:18]: Two more reducers are needed:
* deleteChildren
* resetTree

File: apps/frontend/src/store/previewTree/previewTreeSlice.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const previewTreeSlice = createSlice({
+    name: 'previewTree',
+    initialState: {
+        value: ['ded']
+    },
+    reducers: {
```
Comment (YacineSteeve) [inline apps/frontend/src/store/previewTree/previewTreeSlice.ts:8]: I thought your configuration would be more...working.

File: apps/frontend/src/store/store.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import previewTreeReducer from './previewTree/previewTreeSlice';
+
+export const store = configureStore({
+    reducer: {
+        previewTree: previewTreeReducer
+    }
+});
+
+export type AppDispatch = typeof store.dispatch;
+export type RootState = ReturnType<typeof store.getState>;
```
Comment (YacineSteeve) [inline apps/frontend/src/store/store.ts:11]: Why not just `State`, instead of `RootState` ?

**Review by YacineSteeve — CHANGES_REQUESTED** [review]

File: apps/frontend/src/store/store.ts (hunk 5 above)
Comment (epoundor) [reply apps/frontend/src/store/store.ts:11]: I copy the docs

File: apps/frontend/src/store/previewTree/previewTreeSlice.ts (hunk 4 above)
Comment (epoundor) [reply apps/frontend/src/store/previewTree/previewTreeSlice.ts:8]: What??
It's just a mock data

File: apps/frontend/src/store/previewTree/previewTreeSlice.ts (hunk 3 above)
Comment (epoundor) [reply apps/frontend/src/store/previewTree/previewTreeSlice.ts:18]: Bla bla bla just approve

File: apps/frontend/src/views/Preview/Preview.tsx (hunk 2 above)
Comment (epoundor) [reply apps/frontend/src/views/Preview/Preview.tsx:38]: 😡

File: apps/frontend/src/reportWebVitals.ts (hunk 1 above)
Comment (epoundor) [reply apps/frontend/src/reportWebVitals.ts:3]: Frère je vais te goumer inh

**Review by YacineSteeve — APPROVED** [review]
Wise boy

**Pull request merged**